### PR TITLE
Contracts Tab Scaffolding - Flow Fixes

### PR DESCRIPTION
**Pull request by wbobeirne**
I busted this out into a separate PR as I wanted to keep all flow-related discussion of how we deal with flow / redux / sagas in one thread. I cleaned up the mess I had in #70, but I wanted to make sure I did this the "right" way.

I think the only majorly contentious thing in here is changing the config reducer to `action: Object` where it was previously `action: ConfigAction`. However, reducers receive all actions, not just the ones from that reducer, so the previous way of defining all actions would not have worked in a reducer that listens to actions meant for other reducers anyway.

### CODE REVIEW

**Comment by wbobeirne** [comment]
I forgot to run tests, which hadn't been updated to reflect the action changes I made. I'll circle back to that soon.

File: .flowconfig [hunk 1]
```diff
@@ -7,3 +7,4 @@
 [options]
 module.system.node.resolve_dirname=node_modules
 module.system.node.resolve_dirname=common
+module.name_mapper='.*\(.sass|scss|less|css\)' -> 'empty/object'
```
Comment (crptm) [inline .flowconfig]: fixed in my branch w/o dep, i will resolve when merging

**Comment by wbobeirne** [comment]
From slack discussion: I'll be removing action constants as a part of this. So changing config back to the way it was, and changing contracts to use strings instead of constants.

**Comment by crptm** [comment]
Dont really want to repeat other PR comments, but can we not store contracts in redux for now please

**Comment by wbobeirne** [comment]
Closing this out in favor of the main contracts PR. I think I'm on top of the flow stuff now.

**Pull request closed without merging**